### ISSUE TITLE
[Paper Note] Autonomous Data Selection with Zero-shot Generative Classifiers for Mathematical Texts, Yifan Zhang+, ACL'25 Findings, 2024.02

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#4004]
# URL
- https://arxiv.org/abs/2402.07625
# Authors
  - Yifan Zhang
  - Yifan Luo
  - Yang Yuan
  - Andrew C Yao
# Abstract
  - We present Autonomous Data Selection (AutoDS), a method that leverages base language models themselves as zero-shot "generative classifiers" to automatically curate high-quality mathematical texts. Unlike prior approaches that require human annotations or training a dedicated data filter, AutoDS relies solely on a model's logits to determine whether a given passage is mathematically informative and educational. By integrating AutoDS into a continual pretraining pipeline, we substantially boost downstream performance on challenging math benchmarks (MATH, GSM8K, and BBH) while using far fewer tokens than previous methods. Empirically, our approach achieves roughly a twofold improvement in pretraining token efficiency over strong baselines, underscoring the potential of self-directed data selection in enhancing mathematical reasoning. We release our curated AutoMathText dataset to facilitate future research in automated domain-specific data curation. The AutoMathText dataset is available at https://huggingface.co/datasets/math-ai/AutoMathText. The code is available at https://github.com/yifanzhang-pro/AutoMathText.
# Translation (by gpt-4o-mini)
- 自律的データ選択（AutoDS）を提案します。これは、基本的な言語モデル自体をゼロショットの「生成分類器」として活用し、高品質な数学テキストを自動的にキュレーションする手法です。従来のアプローチが人間の注釈や専用のデータフィルターのトレーニングを必要とするのに対し、AutoDSはモデルのロジットのみに依存して、特定のパッセージが数学的に有益で教育的であるかどうかを判断します。AutoDSを継続的な事前トレーニングパイプラインに統合することで、以前の手法よりもはるかに少ないトークンを使用しながら、難しい数学ベンチマーク（MATH、GSM8K、BBH）での下流性能を大幅に向上させます。実証的に、我々のアプローチは強力なベースラインに対して事前トレーニングのトークン効率を約2倍改善し、数学的推論を強化する自己指向のデータ選択の可能性を強調しています。自動化されたドメイン特化型データキュレーションに関する今後の研究を促進するために、キュレーションされたAutoMathTextデータセットを公開します。AutoMathTextデータセットは、https://huggingface.co/datasets/math-ai/AutoMathText で入手可能です。コードは、https://github.com/yifanzhang-pro/AutoMathText で入手できます。
# Summary (by gpt-4o-mini)
- 自律的データ選択（AutoDS）は、言語モデルをゼロショットの生成分類器として利用し、高品質な数学テキストを自動キュレーションする手法です。従来の方法と異なり、人間の注釈やデータフィルターのトレーニングを必要とせず、モデルのロジットに基づいて数学的に有益なパッセージを判断します。AutoDSは事前トレーニングパイプラインに統合され、数学ベンチマークでの性能を大幅に向上させ、トークン効率を約2倍改善しました。さらに、キュレーションされたAutoMathTextデータセットを公開し、今後の研究を促進します。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/yifan_zhang_/status/2001708613698507039?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
以下のようなzero-shotのmeta-promptを用いてテキストをスコアリングし（Q1, Q2それぞれについてスコア(=logits)を算出し乗算）継続事前学習に利用することで性能が向上することを示した研究。
<img width="393" height="516" alt="Image" src="https://github.com/user-attachments/assets/6f8a5e5c-8ffa-46e2-9ffe-7440ba9130b2" />

ベースライン:
- uniform: OpenWebMathから一様サンプリングする
- DSIR: source dataとtarget domain（今回はPile's Wikipedia splitを利用）のKL Divergenceを比較しデータを選択する。
- Qurating: Reward-modelをベースにした学習サンプルに対するeducational valueをランキングさせる手法

提案手法は
- OpenWebMath
- arXiv (from RedPajama)
- Algebraic Stack
の中からトップスコアのドキュメントを利用。DSIR, Quratingについてはデータソースが明示されていないが、おそらく提案手法揃えていると思われる。また学習する際のトークン量も手法間で（明示的に書かれていないように見えるが）同等にそろえていると思われる。

まずpreliminary experimentsとしてトークン数のbudgetを小さめにして実験。uniformと比較すると、別のmathドメインデータでFinetuningした後のパフォーマンスが向上している。トークン数のbudgetもexactに揃えられている。
<img width="777" height="174" alt="Image" src="https://github.com/user-attachments/assets/5a777fa5-202e-4227-84e7-d383590ae737" />

続いてトークンのbudgetを増やして、~2.5Bトークンにスケールアップして比較（継続事前学習→1 epoch SFT）。提案手法が全体的にdownstreamタスクでの評価で高い性能を発揮。しかしこちらでは、いくつかでuniformの性能もよい。
<img width="784" height="397" alt="Image" src="https://github.com/user-attachments/assets/6a4f0321-476b-49c4-b58a-4cc70604228e" />

また、最後に数学データでの継続事前学習が異なるドメインに対してどの程度転移するかを測ると、提案手法が平均して最もよかった。しかしこちらもでもuniformが結構強い結果に見える。
<img width="806" height="459" alt="Image" src="https://github.com/user-attachments/assets/5cf0bf34-07d7-4ff7-9468-81c7f8e253e1" />

OpenWebMathがそもそもheuristicsとtrained classifierを用いてキュレーションされたデータとのことなので、ある程度高品質であることが想定される。
